### PR TITLE
audio/video media

### PR DESCRIPTION
**Pull request by eelfroth**
this embeds audio/video/img based on file extension. `.ogg` is a problem in this case, because it could contain audio or video.

go to my portal to see this change in action: dat://5276ca2f92b6d508933ac4b01ef9078938f69bed9ca15c462617f531641b7a6b/

### CODE REVIEW

**Comment by frozenpandaman** [comment]
oh god

don't make autoplay a thing

**Comment by eelfroth** [comment]
@frozenpandaman: it doesn't autoplay though? you have to click the play button

File: scripts/entry.js [hunk 1]
```diff
@@ -70,7 +70,19 @@ function Entry(data)
     if(this.media){
       var parts = this.media.split(".")
       if (parts.length === 1) { this.media += ".jpg" } // support og media uploads
-      html += "<img class='media' src='"+this.dat+"/media/content/"+this.media+"'/>";
+      switch(parts[1]){
```
Comment (forresto) [inline scripts/entry.js]: For `my.vid.mp4` this should be `parts[parts.length-1]` ?

Comment (eelfroth) [reply scripts/entry.js]: oh, well spotted, thanks!

**Comment by eelfroth** [comment]
making a version of this that works for any file extension, displayable or not. undisplayable files will get added as download-link.

**Comment by eelfroth** [comment]
now, if we could also embed text-files, that would be great. but that seems less straightforward to implement so I'll do it some other time…

File: scripts/entry.js [hunk 2]
```diff
@@ -69,8 +69,18 @@ function Entry(data)
     var html = "";
     if(this.media){
       var parts = this.media.split(".")
-      if (parts.length === 1) { this.media += ".jpg" } // support og media uploads
-      html += "<img class='media' src='"+this.dat+"/media/content/"+this.media+"'/>";
+      extension = parts[parts.length-1];
```
Comment (forresto) [inline scripts/entry.js]:
``` js
extension = parts[parts.length-1].toLowerCase();
```

for `FILE.JPG` to work

File: scripts/entry.js [hunk 3]
```diff
@@ -69,8 +69,18 @@ function Entry(data)
...
+      if (parts.length === 1) {
+        this.media += ".jpg";
+        extension = "jpg";
+      } // support og media uploads
+      audiotypes = ["mp3", "ogg", "wav"];
+      videotypes = ["mp4", "webm"]; // "ogg",
+      imagetypes = ["apng", "bmp", "dib", "gif", "jpg", "jpeg", "jpe", "png", "svg", "svgz", "tiff", "tif", "webp"];
+      if(audiotypes.indexOf(extension) > -1){ html += "<audio class='media' src='"+this.dat+"/media/content/"+this.media+"' controls />"; }
+      else if(videotypes.indexOf(extension) > -1){ html += "<video class='media' src='"+this.dat+"/media/content/"+this.media+"' controls />"; }
+      else if(imagetypes.indexOf(extension) > -1){ html += "<img class='media' src='"+this.dat+"/media/content/"+this.media+"'/>"; }
```
Comment (forresto) [inline scripts/entry.js]: `if( audiotypes.includes(extension) )` is nice for these checks.

**Comment by forresto** [comment]
> embed text-files

`iframe` could work, for something very quick & dirty... would beaker render formatted `.md` in an iframe as well?

File: scripts/entry.js (hunk 3 above)
Comment (eelfroth) [reply scripts/entry.js]: `foo.indexOf(bar) > -1` is the method used everywhere else in the codebase, so I think I'll keep it.

File: scripts/entry.js (hunk 2 above)
Comment (eelfroth) [reply scripts/entry.js]: done

**Comment by eelfroth** [comment]
> iframe could work

I'll look into that. need sleep first :)

> would beaker render formatted .md in an iframe

doesn't appear to be the case, sadly

**Comment by eelfroth** [comment]
hmm, `iframe` could have security issues, i.e. somebody embeds an html that loads malicious javascript.

File: scripts/entry.js [hunk 4]
```diff
@@ -69,8 +69,18 @@ function Entry(data)
...
+      else{ html +="<a class='media' href='"+this.dat+"/media/content/"+this.media+"'>&gt;&gt; "+this.media+"</a>"; }
```
Comment (forresto) [inline scripts/entry.js]: Fine default 😄

**Comment by ianh** [comment]
We should probably escape "this.media" to avoid the security issues with unescaped string replacement.  The existing code has the same issues, though, so might as well land this, thanks!

**Pull request merged**